Zend/zend_exceptions.c

```diff
@@ -391,106 +396,181 @@ ZEND_METHOD(error_exception, __construct)
 }
 /* }}} */
 
+#define GET_OBJ_PROPERTY(object, id, silent) \
+	zend_obj_read_property_ex(_obj_get_exception_base(object), (object), ZSTR_KNOWN(id), silent, &rv)
+
 #define GET_PROPERTY(object, id) \
 	zend_read_property_ex(i_get_exception_base(object), (object), ZSTR_KNOWN(id), 0, &rv)
 #define GET_PROPERTY_SILENT(object, id) \
 	zend_read_property_ex(i_get_exception_base(object), (object), ZSTR_KNOWN(id), 1, &rv)
 
+static zend_long *_get_long_prop(zend_object *exception, zend_known_string_id id, zend_bool silent)
+{ /* {{{ */
+	zval rv;
+	zval *prop = GET_OBJ_PROPERTY(exception, id, 1);
+	ZVAL_DEREF(prop);
+	if (Z_TYPE_P(prop) != IS_LONG) {
+		if (!silent) {
+			zend_type_error("Exception %s was not a long", ZSTR_VAL(ZSTR_KNOWN(id)));
```

Comment:
Might it make more sense to switch these to use typed properties, so this is enforced with the now built-in mechanism? Seems like something we can do in PHP 8...

Reply:
Definitely, it makes no sense to check types during property reading. The wrong values must not be written there at all.

Reply:
Can typed properties be subverted via reflection?

Reply:
The problem is certainly PDOException here where code is a string. Same could be true for any other userland exception which just writes to $this->code. Changing code to a typed string is a serious BC break.